core/src/main/scala/kafka/admin/ReassignPartitionsCommand.scala

```diff
@@ -500,7 +526,7 @@ class ReassignPartitionsCommand(zkUtils: ZkUtils,
 
   /** Set throttles to replicas that are moving. Note: this method should only be used when the assignment is initiated. */
   private[admin] def assignThrottledReplicas(allExisting: Map[TopicAndPartition, Seq[Int]], allProposed: Map[TopicAndPartition, Seq[Int]], admin: AdminUtilities = AdminUtils): Unit = {
-    for (topic <- allProposed.keySet.map(_.topic).toSeq) {
+    for (topic <- allProposed.keySet.map(_.topic).toSeq.distinct) {
```

Comment:
`distinct` is called several times, it has performance penalty, is it possible to use Set?